packages/components/namespace-store/namespace-store-form.tsx

```diff
@@ -0,0 +1,279 @@
+        ...nsPayload.spec,
+        [PROVIDERS_NOOBAA_MAP[provider]]: {
+          [BUCKET_LABEL_NOOBAA_MAP[provider]]: providerDataState.target,
+          secret: {
+            name: secretName,
+            namespace
+          }
+        }
+      };
+    }
+    if (provider === BC_PROVIDERS.S3) {
+      nsPayload.spec.s3Compatible = {
+        ...nsPayload.spec.s3Compatible,
+        endpoint: providerDataState.endpoint
+      };
+    } else if (provider === BC_PROVIDERS.IBM) {
+      nsPayload.spec.ibmCos = {
+        ...nsPayload.spec.ibmCos,
+        endpoint: providerDataState.endpoint
+      };
+    }
+    // Add region in the end
+    if (provider === BC_PROVIDERS.AWS) {
+      nsPayload.spec.awsS3 = {
+        ...nsPayload.spec.awsS3,
+        region: providerDataState.region
+      };
+    }
+    if (provider === BC_PROVIDERS.FILESYSTEM) {
+      nsPayload.spec.nsfs = {
+        ...nsPayload.spec.nsfs,
+        pvcName: pvc,
+        subPath: folderName
+      };
+    }
+    promises.push(
+      k8sCreate({ model: NooBaaNamespaceStoreModel, data: nsPayload })
+    );
+    return handlePromise(
+      Promise.all(promises),
+      (resources: (NamespaceStoreKind | SecretKind)[]) => {
+        redirectHandler(resources);
+      }
+    );
+  };
+
+  return (
+    <Form
+      className={classNames("nb-endpoints-form", "co-m-pane__body", className)}
+      onSubmit={onSubmit}
+      noValidate={false}
+    >
+      <FormGroup
+        label={t("Namespace store name")}
+        fieldId="namespacestore-name"
+        className="nb-endpoints-form-entry"
+        helperText={t(
+          "A unique name for the namespace store within the project"
+        )}
+        isRequired
+      >
+        <Tooltip
+          content={t("Name can contain a max of 43 characters")}
+          isVisible={nsName.length > 42}
+          trigger="manual"
+        >
+          <TextInput
+            id="ns-name"
+            onChange={handleNsNameTextInputChange}
+            value={nsName}
+            maxLength={43}
+            data-test="namespacestore-name"
+            placeholder="my-namespacestore"
+          />
+        </Tooltip>
+      </FormGroup>
+
+      <FormGroup
+        label={t("Provider")}
+        fieldId="provider-name"
+        className="nb-endpoints-form-entry"
+        isRequired
+      >
+        <SingleSelectDropdown
+          id="providers"
+          className="nb-endpoints-form-entry__dropdown"
+          onChange={(selectedProvider) => {
+            setProvider(PROVIDERS[selectedProvider]);
+          }}
+          selectOptions={providerDropdownOptions(t)}
+          selectedKey={provider}
+        />
+      </FormGroup>
+      {(provider === BC_PROVIDERS.AWS ||
+        provider === BC_PROVIDERS.S3 ||
+        provider === BC_PROVIDERS.IBM ||
+        provider === BC_PROVIDERS.AZURE) && (
+        <S3EndPointType
+          type={StoreType.NS}
+          provider={provider}
+          namespace={null}
```

Comment:
why null ??
plz when copying keep it same as it was in console code. Pass DATA_FEDERATION_NS here.